core/src/main/scala/org/ensime/indexer/SearchService.scala

```diff
@@ -64,85 +67,85 @@ class SearchService(
 
     val jarUris = config.allJars.map(vfs.vfile).map(_.getName.getURI)
 
-    // remove stale entries: must be before index or INSERT/DELETE races
-    val stale = for {
-      known <- db.knownFiles()
-      f = known.file
-      name = f.getName.getURI
-      if !f.exists || known.changed ||
-        (name.endsWith(".jar") && !jarUris(name))
-    } yield f
-
-    log.info(s"removing ${stale.size} stale files from the index")
-    if (log.isTraceEnabled)
-      log.trace(s"STALE = $stale")
-
-    // individual DELETEs in H2 are really slow
-    val removing = stale.grouped(1000).map { files =>
-      Future {
-        index.remove(files)
-        db.removeFiles(files)
-      }(workerEC)
-    }
-
-    val removed = Future.sequence(removing).map(_ => stale.size)
+    for {
```

Comment:
cool. How do you suggest handling potential history re-writes here and integration of those changes?